### PR TITLE
feat: add link to google translate in publications

### LINKED ISSUE #2755: Add link to google translate in publications

### PR DESCRIPTION
**Pull request by foolo**
## What does this PR do?

Add link to Google Translate in publications.
@bigint Let me know if I have understood the intention of the task :)

### <samp>🤖 Generated by Copilot at 7bbbd18</samp>

Added a feature to translate publications using Google Translate. The feature uses a new function `getPrimaryLanguage` in `i18n.ts` to detect the language of the publication and the user interface.

## Screen recording

https://user-images.githubusercontent.com/667227/235431540-9d0ce3ae-df57-4d01-a7c6-565c4a2d18bd.mp4

## Related issues

Fixes #2755

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at 7bbbd18</samp>

*  Add translation functionality to `PublicationBody` component ([link](https://github.com/lensterxyz/lenster/pull/2768/files?diff=unified&w=0#diff-cf87c96f430fb044b052837b8732058bdc59aba03cba72f7f7c43143622ffae1L6-R8), [link](https://github.com/lensterxyz/lenster/pull/2768/files?diff=unified&w=0#diff-cf87c96f430fb044b052837b8732058bdc59aba03cba72f7f7c43143622ffae1L23-R32), [link](https://github.com/lensterxyz/lenster/pull/2768/files?diff=unified&w=0#diff-cf87c96f430fb044b052837b8732058bdc59aba03cba72f7f7c43143622ffae1R46-R52), [link](https://github.com/lensterxyz/lenster/pull/2768/files?diff=unified&w=0#diff-cf87c96f430fb044b052837b8732058bdc59aba03cba72f7f7c43143622ffae1R83-R92))
*  Define `getPrimaryLanguage` function in `i18n.ts` module ([link](https://github.com/lensterxyz/lenster/pull/2768/files?diff=unified&w=0#diff-a0b11c1dcd224a0790b465ed1cf240b02162b537383602c72fcbc02eb3e6d688R47-R57))

## Emoji

🌐🧮🔗

### CODE REVIEW

**Comment by bigint** [comment]
I think we can do something like bluesky does, nothing fancy!

Screenshot 2023-05-01 at 7 56 16 PM

After clicking it takes to

```
https://translate.google.com/?sl=auto&tl=en&text=as%20a%20kid%20i%20loved%20reading%20Great%20Soviet%20Encyclopedia.%20it%20was%20split%20in%2030%20volumes.%20but%20wait%20%E2%80%94%20if%20you%20count%20them%2C%20it%E2%80%99ll%20be%2031!%20that%E2%80%99s%20because%20volume%2024%20is%20split%20into%20two%20books.%2024i%20is%20just%20words%20starting%20with%20%C2%AB%D0%A1%C2%BB%20but%2024ii%20is%20an%20entire%20tome%20for%20a%20singular%20word&op=translate
```

**Comment by bigint** [comment]
@foolo

**Comment by foolo** [comment]
@bigint Thanks for the feedback. I updated the design a bit.
See recording:

https://user-images.githubusercontent.com/667227/235477832-5134f359-6563-4d4c-b7ee-748e6d443d97.mp4

**Comment by foolo** [comment]
@bigint Should I add a new tracking data event, such as `Mixpanel.track(PUBLICATION.TRANSLATE);` ?

**Comment by bigint** [comment]
@foolo Can't we make it auto detect on google translate itself?

I think we don't need to use linui for locale here 🤔

**Comment by bigint** [comment]
Also updated your GH invitation to repo level 🙏🏼 just for some compliances 🙇🏼

File: apps/web/src/components/Publication/Actions/Menu/Translate.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        )
+      }
+      href={getGoogleTranslateUrl(publication?.metadata?.content)}
+      onClick={(event) => {
+        stopEventPropagation(event);
+      }}
+      target="_blank"
+    >
+      <div className="flex items-center space-x-2">
+        <GlobeAltIcon className="h-4 w-4" />
```
Comment (bigint) [inline apps/web/src/components/Publication/Actions/Menu/Translate.tsx:41]: We can use translate icon 🙇🏼 

image

**Comment by foolo** [comment]
> @foolo Can't we make it auto detect on google translate itself?
> 
> I think we don't need to use linui for locale here thinking

Yep, we can probably **always** show Translate in the menu, and everything will be much more simple.

Details:
Google translate is actually already used for detection ( `/?sl=auto` in the URL).
Lingui ( `i18n.locale` ) is used to get the user's selected UI language, which is then compared to the language of the publication (`publication?.metadata?.locale`), and if they are the same, we hide the "Translate" option in the menu.

But: I just realized that `publication?.metadata?.locale` is not reliable. It may often show the wrong language if the author is multilingual. So the best and most simple is probably to remove the use of both `i18n.locale` and `publication?.metadata?.locale` and  **always** show the Translate option (If the user for some reason wants to translate from Spanish to Spanish, they may do so 🙂)

**Comment by bigint** [comment]
Yup exactly @foolo 🙇🏼

**Comment by bigint** [comment]
I updated the PR!

LGTM now 🚀

**Comment by foolo** [comment]
> I updated the PR!
> 
> LGTM now rocket

@bigint I see that now the behavior is a bit different. Now it always translates to English (and not to the user's UI language as before). Was that intentional? Just asking in case it was a mistake.

(What I meant by _"remove the use of both i18n.locale and publication?.metadata?.locale and always show the Translate option"_ was to remove it for determining whether we should show the Translate option, not removing it for determining the target language)

**Pull request merged**